Products/ZenHub/zenhub.py

```diff
@@ -993,11 +780,203 @@ def buildOptions(self):
+        return pb.IPerspective, self.hubAvitar, lambda: None
+
+
+class HubAvitar(pb.Avatar):
+    """
+    Connect collectors to their configuration Services
+    """
+
+    def __init__(self, hub):
+        self.hub = hub
+
+    def perspective_ping(self):
+        return 'pong'
+
+    def perspective_getHubInstanceId(self):
+        return os.environ.get('CONTROLPLANE_INSTANCE_ID', 'Unknown')
+
+    def perspective_getService(self,
+                               serviceName,
+                               instance=None,
+                               listener=None,
+                               options=None):
+        """
+        Allow a collector to find a Hub service by name.  It also
+        associates the service with a collector so that changes can be
+        pushed back out to collectors.
+
+        @type serviceName: string
+        @param serviceName: a name, like 'EventService'
+        @type instance: string
+        @param instance: the collector's instance name, like 'localhost'
+        @type listener: a remote reference to the collector
+        @param listener: the callback interface to the collector
+        @return a remote reference to a service
+        """
+        try:
+            service = self.hub.getService(serviceName, instance)
+        except RemoteBadMonitor:
+            # This is a valid remote exception, so let it go through
+            # to the collector daemon to handle
+            raise
+        except Exception:
+            self.hub.log.exception("Failed to get service '%s'", serviceName)
+            return None
+        else:
+            if service is not None and listener:
+                service.addListener(listener, options)
+            return service
+
+    def perspective_reportingForWork(self, worker, workerId):
+        """
+        Allow a worker register for work.
+
+        @type worker: a pb.RemoteReference
+        @param worker: a reference to zenhubworker
+        @return None
+        """
+        worker.busy = False
+        worker.workerId = workerId
+        self.hub.log.info("Worker %s reporting for work", workerId)
+        self.hub.workers.append(worker)
+        self.hub.updateEventWorkerCount()
+
+        def removeWorker(worker):
+            if worker in self.hub.workers:
+                self.hub.workers.remove(worker)
+                self.hub.updateEventWorkerCount()
+                self.hub.log.info("Worker %s disconnected", worker.workerId)
+
+        worker.notifyOnDisconnect(removeWorker)
+
+
+HubWorklistItem = collections.namedtuple(
+    'HubWorklistItem',
+    'recvtime deferred servicename instance method args'
+)
+WorkerStats = collections.namedtuple(
+    'WorkerStats',
+    'status description lastupdate previdle'
+)
+LastCallReturnValue = collections.namedtuple(
+    'LastCallReturnValue', 'returnvalue'
+)
+
+try:
+    NICE_PATH = subprocess.check_output('which nice', shell=True).strip()
+except Exception:
+    NICE_PATH = None
```

Comment:
The NICE_PATH assignments can be removed.  They're not used anywhere.

Reply:
Removed the unused NICE_PATH code, and rebased the commit